src/lib/dirsize.c

```diff
@@ -96,6 +97,13 @@ double libreport_get_dirsize_find_largest_dir(
             log_debug("Skipping %s': sosreport is being generated.", dname);
             goto next;
         }
+        if (lstat(libreport_concat_path_file(dname, ".lock"), &statbuf) == 0)
```

Comment:
Aren’t you leaking the string here?

Reply:
Could we also use `g_build_filename()` here?

Reply:
> Aren’t you leaking the string here?  
  
`dname`? That one is `free()`'d at `next:`.

Reply:
> Could we also use `g_build_filename()` here?  
  
Oh, absolutely, but I'd rather do it in a separate pull request along with all the other GLib migration changes.

Reply:
> Oh, absolutely, but I'd rather do it in a separate pull request along with all the other GLib migration changes.

Okay, just throwing it out there to make it one line less when you do that.

Reply:
> `dname`? That one is `free()`'d at `next:`.

`libreport_concat_path_file()` returns a heap-allocated string, which should be freed.

Reply:
Leaks fixed.